python/rikai/spark/functions/__init__.py

```diff
@@ -15,6 +15,22 @@
 """Domain-specific Pyspark UDFs
 """
 
+from pyspark.sql import SparkSession
+
 from rikai.spark.functions.geometry import *
 from rikai.spark.functions.io import *
 from rikai.spark.functions.vision import *
+
+__all__ = ["init"]
+
+
+def init(spark: SparkSession):
+    """Register all rikai UDFs"""
+    from rikai.spark.functions import geometry, io, vision
+
+    for name in geometry.__all__:
```

Comment:
Shall we make sure that these exported methods are actually spark udf?

Reply:
I do not think so. Because the set of Rikai built-in is a finite set!

Let it crash when it is not actually a spark udf.

Reply:
I've checked the source code of PySpark. There are no ready-to-use functions to check if a python function is a valid PySpark udfs.

This pr assumes the public methods defined in `__all__` for these this python files should be valid pyspark udf.

If one of them is not valid, it will be found during Github Action.

Reply:
i think it's likely that we'll have a larger set of UDFs than what we want to always register at init. "explicit is better than implicit" after all.

Reply:
I guess what I meant is that, this PR prevents `geometry` and etc modules to export any non `udf` functions/classes in `__all__`.

Reply:
how about add a new global variable `__all_udfs__` for this case?

Reply:
How about:
1. we create a `RikaiUdfRegistry` singleton class that contains an internal dict of {name -> udf func}
2. we create a `@rikai_udf` decorator so that it creates a spark udf and also registers the function in the registry
3. within this `init` function, just call `registry.register_all()`

@da-tubi @eddyxu wdyt?